lib/src/api/trace/span_link.dart

```diff
@@ -7,7 +7,7 @@ class SpanLink {
   final api.SpanContext _context;
   final List<api.Attribute> _attributes;
 
-  SpanLink(this._context, {List<api.Attribute> attributes})
+  SpanLink(this._context, {List<api.Attribute>? attributes})
       : _attributes = attributes ?? [];
```

Comment:
In dart, does having a question mark on the optional named parameter mean that doing `new SpanLink(ctx, {attributes: null})` is valid? Can we avoid this by doing something like the following?

```suggestion
  SpanLink(this._context, {List<api.Attribute> attributes = const []})
      : _attributes = new List<api.Attribute>()..addAll(attributes);
```

Reply:
Oh, I see that you may want to be able to add attributes later which using a const list would prevent. Feel free to disregard my suggestion.